pytorch_lightning/trainer/optimizers.py

```diff
@@ -54,15 +59,25 @@ def init_optimizers(self, model: LightningModule) -> Tuple[List, List, List]:
         # multiple dictionaries
         elif isinstance(optim_conf, (list, tuple)) and all(isinstance(d, dict) for d in optim_conf):
             optimizers = [opt_dict["optimizer"] for opt_dict in optim_conf]
-            lr_schedulers = [opt_dict["lr_scheduler"] for opt_dict in optim_conf if "lr_scheduler" in opt_dict]
+            scheduler_dict = (
+                lambda scheduler, opt_idx: dict(scheduler, opt_idx=opt_idx)
+                if isinstance(scheduler, dict)
+                else dict({'scheduler': scheduler, 'opt_idx': opt_idx})
```

Comment:
Does this work? If the else is executed, line 69 will fail because dicts are not callable

Reply:
Also, this function is about parsing the user input, what you are trying to do is probably better suited to the function below (`configure_schedulers`)

Reply:
I think this should be fine when calling on Line 69 cause it is wrapped in a lambda function. The reason I set the `opt_idx` here and not in `configure_schedulers` is because `configure_schedulers` just takes in an array of lr_schedulers. So without assigning an opt_idx, a lr_scheduler in `configure_schedulers` won't have the knowledge of which optimizer it is associated with.